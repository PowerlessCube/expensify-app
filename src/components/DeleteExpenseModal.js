import React from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal';
import { hideModal } from '../actions/modal';
import { startRemoveExpense } from '../actions/expenses';

export class DeleteExpenseModal extends React.Component {
   
    onRemove = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    }

    hideModal = () => {
        this.props.hideModal();
    }

    render() {
        return (
            <Modal
                isOpen={this.props.modal.isOpen}
                contentLabel="Delete Expense"
            >
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Delete {this.props.expense.name}?</h1>        
                    </div>
                </div>
                <div className="content-container">
                    <button 
                        className="button button--secondary"
                        onClick={this.onRemove}
                    >Remove Expense</button>
                    <button 
                        className="button button--secondary"
                        onClick={this.hideModal}
                    >Cancel</button>
                </div>
            </Modal>
        );
    }
}

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id),
    modal: state.modal
});

const mapDispatchToProps = (dispatch, props) => ({
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data)),
    hideModal: () => dispatch(hideModal())
});

export default connect(mapStateToProps, mapDispatchToProps)(DeleteExpenseModal);