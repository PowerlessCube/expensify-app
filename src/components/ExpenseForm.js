import React from 'react';
//Installed moment - a handy date utility library for js
import moment from 'moment';
//Installed airbnb datepicker that requires moment
import {SingleDatePicker} from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';

// const date = new Date();
const now = moment();
console.log(now.format('Do MMM, YYYY'));

class ExpenseForm extends React.Component {
    state = {
        description: '',
        note: '',
        amount: '',
        // initial value
        createdAt: moment(),
        calendarFocused: false
    };

    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    };

    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState({ note });
    }   

    onAmountChange = (e) => {
        const amount = e.target.value;
        if (amount.match(/^\d*(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount }))
        }
    }

    onDateChange = (createdAt) => {
        this.setState(() => ({ createdAt }));
    }

    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }))
    }

    render() {
        return (
            <div>
                <form action="">
                    <input 
                        type="text"
                        placeholder="Description"
                        autoFocus
                        value={this.state.description}
                        onChange={this.onDescriptionChange}
                    />
                    <input 
                        type="text"
                        placeholder="Amount"
                        value={this.state.amount}
                        onChange={this.onAmountChange}
                    />
                    <SingleDatePicker 
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={(day) => false}
                    />
                    <textarea 
                        placeholder="Add a note for your expense (optional)"
                        value={this.state.note}
                        onChange={this.onNoteChange}
                    >
                    </textarea>
                    <button>Add Expense</button>
                </form>
            </div>
        );
    }
}

export default ExpenseForm;