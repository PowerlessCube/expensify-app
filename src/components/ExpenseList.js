import React from 'react';
// We call connect
import { connect } from 'react-redux';

// Regular unconnected component
const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        {props.expenses.length}
    </div>
);

// We define the things that we want to get form the store
// Some functions
const mapStateToProps = (state) => {
    return {
        expenses: state.expenses,
        filters: state.filters
    };
}

// Connection to the store
// We get functions back from this so we have to call it. Hence the 

// and we define the component we want to connect to the store
export default connect(mapStateToProps)(ExpenseList);