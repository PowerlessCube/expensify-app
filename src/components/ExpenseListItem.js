import React from 'react';
import { connect } from 'react-redux';

// Export a stateless functional component
const ExpenseListItem = (props) => (
    <div>
        <p>{props.description}</p>
        <p>{props.amount}</p>
        <p>{props.createdAt}</p>
    </div>
);

export default ExpenseListItem;
// description, amount, createdAt