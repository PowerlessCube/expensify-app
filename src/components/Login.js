import React from 'react';

const Login = () => (
    <div>
        <button>Login</button>
    </div>
)

export default Login