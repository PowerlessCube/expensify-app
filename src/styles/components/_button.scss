.button__login {
    background: $google-blue;
    border-radius: 0px;
    color: white;
    padding: $s-size;
    border-color: lighten( $google-blue, 10%);
}