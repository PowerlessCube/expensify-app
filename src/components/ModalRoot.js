import React from 'react';
import { connect } from 'react-redux';
import DeleteExpenseModal from './DeleteExpenseModal';

const MODAL_COMPONENTS = {
    'DELETE_POST': DeleteExpenseModal
}

const ModalRoot = ({ modalType, modalProps }) => {
    if (!modalType) {
        return <span />
    }

    const SpecificModal = MODAL_COMPONENTS[modalType]
    return <SpecificModal {...modalProps} />
}

const mapState = state => state.modal;

export default connect(mapState)(ModalRoot)
