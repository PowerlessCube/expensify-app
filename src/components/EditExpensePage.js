import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startRemoveExpense, startEditExpense } from '../actions/expenses';
import { showModal } from '../actions/modal';

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        console.log(this.props);
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    }
    showModal = (expense) => {
        console.log(this.props);
        this.props.showModal(expense)
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>        
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button 
                        className="button button--secondary"
                        onClick={this.showModal}
                    >Remove Expense</button>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id),
    modal: state.modal
});

const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    showModal: (expense) => dispatch(showModal(expense))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);