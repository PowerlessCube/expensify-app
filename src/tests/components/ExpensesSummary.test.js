import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should render expensesSummary correctly with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={2.35} />)
    expect(wrapper).toMatchSnapshot();
});

test('Should render expensesSummary correctly with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={235} expensesTotal={12341241235421243} />)
    expect(wrapper).toMatchSnapshot();
});