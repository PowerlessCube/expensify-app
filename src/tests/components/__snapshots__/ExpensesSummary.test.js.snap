// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expensesSummary correctly with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $0.02
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should render expensesSummary correctly with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        235
      </span>
       
      expenses
       totalling 
      <span>
        $123,412,412,354,212.44
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
